falcon/request_helpers.py

```diff
@@ -38,6 +42,29 @@ def fget(self):
     return property(fget)
 
 
+def parse_etags(etag_str):
+    """
+    Parse a string of ETags given in an If-Match or If-None-Match header as
+    defined by RFC 7232.
+
+    Args:
+        etag_str (str): A string of ASCII characters placed between double quotes
+            and may be prefixed by ``W/`` to indicate that the weak comparison
+            algorithm should be used.
```

Comment:
It is good to reuse code for both `If-Match` and `If-None-Match`, however, my impression is that the comparison function handling is different according to the RFC [[1]](https://tools.ietf.org/html/rfc7232):

> `If-Match`
> <...> An origin server MUST use the **strong** comparison function when
   comparing entity-tags for If-Match <...>

> `If-None-Match`
> <...> A recipient MUST use the **weak** comparison function when comparing
   entity-tags for If-None-Match <...>

In your proposed changeset, we would present the stripped `If-Match` values to the API user as if they were all suitable for strong comparison, but it may not be the case.

In addition, see also the other comment regarding parsing, we should be crystal clear what we do with weak `ETag`s in this case, since just silently discarding them might also contribute to confusion (not as much as masking parse errors though).

1. https://tools.ietf.org/html/rfc7232